### PR TITLE
 Support added for ECC append information in IP_DATA.

### PR DESCRIPTION
**Pull request by rozumx**
Hello Max and Lizhi,
Could you spend extra time reviewing the changes to struct ip_data in xclbin.h.  

If there are additional changes you wish me to make please don't hesitate to ask.

Thanks, Stephen

### CODE REVIEW

**Comment by maxzhen** [comment]
ok to test

File: src/runtime_src/driver/include/xclbin.h [hunk 1]
```diff
@@ -247,7 +249,14 @@ extern "C" {
     /* IPs on AXI lite - their types, names, and base addresses.*/
     struct ip_data {
         uint32_t m_type; //map to IP_TYPE enum
-        uint32_t properties; //32 bits to indicate ip specific property. eg if m_type == IP_KERNEL then bit 0 is for interrupt.
+        union {
+            uint32_t properties; //32 bits to indicate ip specific property. eg if m_type == IP_KERNEL then bit 0 is for interrupt.
+            struct {     // Used by IP_MEM_* types
+               uint32_t m_index;
```
Comment (maxzhen) [inline src/runtime_src/driver/include/xclbin.h]: do you mean uint16_t here?

**Comment by maxzhen** [comment]
retest this please

File: src/runtime_src/driver/include/xclbin.h (hunk 1 above)
Comment (rozumx) [reply src/runtime_src/driver/include/xclbin.h]: Great catch...yes it needs to be uint16_t.

**Comment by rozumx** [comment]
Hello Max, 
I have updated the code to have m_index be a unit16_t and NOT uint32_t.

**Pull request merged**